### PR TITLE
PR3 - Removing AlicaEngine::this from SyncModule 

### PR DESCRIPTION
**Pull request by triccyx**
Removing AlicaEngine::this from SyncModule and subclasses

### CODE REVIEW

**Comment by triccyx** [comment]
Aligned with rr_devel with a merge

**Review by veerajsk — APPROVED** [review]

File: alica_engine/include/engine/syncmodule/SyncModule.h [hunk 1]
```diff
@@ -39,15 +43,22 @@ class SyncModule
     void sendAcks(const std::vector<SyncData>& syncDataList) const;
     void synchronisationDone(const Synchronisation* st);
 
+    void reload(const YAML::Node& config);
```
Comment (abhi1287) [inline alica_engine/include/engine/syncmodule/SyncModule.h:46]: doesn't need to be public

File: alica_engine/src/engine/syncmodule/SyncModule.cpp [hunk 2]
```diff
@@ -26,25 +26,40 @@ using std::pair;
...
 SyncModule::~SyncModule()
 {
     for (auto iter : _synchProcessMapping) {
         delete iter.second;
     }
 }
+
+void SyncModule::reload(const YAML::Node& config)
+{
+    _maySendMessages = !config["Alica"]["SilentStart"].as<bool>();
```
Comment (abhi1287) [inline alica_engine/src/engine/syncmodule/SyncModule.cpp:54]: [Improvement suggestion] in a later pr, we can get rid of _maySendMessages everywhere and should only have it as part of communication

**Review by abhi1287 — APPROVED** [review]

File: alica_engine/include/engine/syncmodule/SyncModule.h (hunk 1 above)
Comment (veerajsk) [reply alica_engine/include/engine/syncmodule/SyncModule.h:46]: @triccyx please address this comment in any one of your later PRs. I am merging this so we can continue reviewing the other PRs.

**Comment by abhi1287** [comment]
lol, i was just pushing the access change. never-mind.

**Pull request merged**